WasatchNET/EEPROM.cs

```diff
@@ -1387,6 +1421,40 @@ public bool read()
                             intensityCorrectionOrder = 0;
                         }
 
+                        if (format >= 7)
+                        {
+                            avgResolution = ParseData.toFloat(pages[3], 48);
+                        }
+                        else
+                        {
+                            avgResolution = 0.0f;
+                        }
+
+                        if (format >= 8)
+                        {
+                            wavecalCoeffs[4] = ParseData.toFloat(pages[2], 21);
+                            subformat = (PAGE_SUBFORMAT)ParseData.toUInt8(pages[5], 63);
+                            if (subformat == PAGE_SUBFORMAT.USER_DATA)
+                            {
+                                intensityCorrectionOrder = 0;
+                                intensityCorrectionCoeffs = null;
+
+                                userData = new byte[192];
+                                //Array.Copy(pages[4], userData, userData.Length);
+                                Array.Copy(pages[4], 0, userData, 0, 64);
+                                Array.Copy(pages[6], 0, userData, 64, 64);
+                                Array.Copy(pages[7], 0, userData, 128, 64);

```

Comment:
does this work if userData is only 64 or 128 bytes long? (I don't know)

Reply:
I think I'm actually more worried about the write than the read, though it's possible the new write will cause issues with the read?